libsoftwarecontainer/src/gateway/devicenode/devicenodegateway.cpp

```diff
@@ -32,14 +33,28 @@ DeviceNodeGateway::DeviceNodeGateway() :
 ReturnCode DeviceNodeGateway::readConfigElement(const json_t *element)
 {
     DeviceNodeParser parser;
+    DeviceNodeLogic logic;
     DeviceNodeParser::Device dev;
 
     if (isError(parser.parseDeviceNodeGatewayConfiguration(element, dev))) {
         log_error() << "Could not parse device node configuration";
         return ReturnCode::FAILURE;
     }
 
-    m_devList.push_back(dev);
+    auto item = std::find_if(m_devList.begin(), m_devList.end(),
+        [&] (DeviceNodeParser::Device const &d) { return d.name == dev.name; });
+
+    if (item == std::end(m_devList)) {
+        m_devList.push_back(dev);
+    } else {
+        if ((item->major == dev.major) && (item->major == dev.minor)) {
+            item->mode = logic.calculateDeviceMode(item->mode, dev.mode);
+        } else {
+            //now we have a new device with same name which shouldn't be existed
```

Comment:
This could be written in better language. But perhaps this whole method would benefit from an earlier and more descriptive comment on what's going on in general.